src/Main.cpp

```diff
@@ -0,0 +1,177 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+#include <algorithm>
+#include "Location.h"
+#include "DistanceCalculator.h"
+
+std::map<std::string, Location> dataMap;
+
+void readFile() {
+    std::string filePath;
```

Comment:
the logic is handled in the main, consider using more classes, main handles reading inputs, searching and printing results